pulp_2to3_migration/app/plugin/content.py

```diff
@@ -416,46 +425,49 @@ def get_remote_by_importer_id(importer_id):
                             pb.increment()
                         continue
 
+                    relative_path = (
+                        pulp_2to3_detail_content.relative_path_for_content_artifact
+                    )
+                    remote_lce_tuples = []
+                    deferred_download = not pulp2content.downloaded
+
                     if is_lazy_type and pulp2lazycatalog:
+                        for lce in pulp2lazycatalog:
+                            remote = get_remote_by_importer_id(lce.pulp2_importer_id)
+                            if remote:
+                                remote_lce_tuples.append((remote, lce))
+
                         # handle DA and RA creation for content that supports on_demand
                         # Downloaded or on_demand content with LCEs.
                         #
                         # To create multiple remote artifacts, create multiple instances of
                         # declarative content which will differ by url/remote in their
                         # declarative artifacts
-                        at_least_one_lce_migrated = False
-                        for lce in pulp2lazycatalog:
-                            remote = get_remote_by_importer_id(lce.pulp2_importer_id)
-                            deferred_download = not pulp2content.downloaded
-                            if not remote and deferred_download:
-                                continue
 
-                            relative_path = (
-                                pulp_2to3_detail_content.relative_path_for_content_artifact
-                            )
+                    if remote_lce_tuples:
+                        for remote, lce in remote_lce_tuples:
                             da = DeclarativeArtifact(
                                 artifact=artifact,
                                 url=lce.pulp2_url,
                                 relative_path=relative_path,
                                 remote=remote,
                                 deferred_download=deferred_download)
                             lce.is_migrated = True
-                            at_least_one_lce_migrated = True
                             dc = DeclarativeContent(content=pulp3content, d_artifacts=[da])
+                            future_relations.update({'lces': list(pulp2lazycatalog)})
```

Comment:
i think this update should happen  outside the for loop

Reply:
It used to be and I can't get my head around why.
It might be a bit redundant but at least LCE will be updated.
When it is outside of the loop, I'm not sure how correct LCEs go to the next stage if we are assigning them to the future_relations after we pushed content to the queue for the next stage.

Reply:
after a short call we had, it was decided to leave these changes as they are and leave a comment in the code. The confusion was around assigning in the future_relations all lazy catalog to the dc instead of one single lce.